R/module_nested_tabs.R

```diff
@@ -307,23 +304,16 @@ srv_nested_tabs.teal_module <- function(id, datasets, modules, is_module_specifi
   )
 
   hashes <- calculate_hashes(datanames, datasets)
-  metadata <- lapply(datanames, datasets$get_metadata)
-  names(metadata) <- datanames
 
-  new_tdata(
+  teal.data::new_teal_data(
```

Comment:
if data is a list of eventReactives then this constructor doesn't work

```
  data <- sapply(
    datanames,
    function(x) eventReactive(trigger_data(), datasets$get_data(x, filtered = TRUE)),
    simplify = FALSE
  )
```

Reply:
This constructor works on `FilteredData`, does it not?

Reply:
`teal.data::new_teal_data` has no dependency on `FIlteredData` - it is generic constructor of teal_data class. In your code here `teal.data::new_teal_data` uses `data` which is a list of reactives (see like 300). This is destined to fail. This is why I suggested to remove `trigger_data` from this function and use it in a parent frame like below

```
data <- eventReactive(trigger_data(), .datasets_to_data(modules, datasets))
```

Reply:
I clearly misunderstand. We are talking about `datasets_to_data` which converts `FilteredData` to `teal_data`, yes?

Reply:
Yes, we are discussing `.datasets_to_data`.  Below lines 300-309, let's go through these calls:

- data is a list of reactives which return filtered data
- hashed - let's ignore this
- teal.data::new_teal_data(data, ...) - call to create teal_data object from list of reactives. Call itself probably wouldn't throw because we don't care what developer pushes to qenv (there is no assert on dataset). Instead of passing data.frames you are passing reactives ;]


```r
data <- sapply(
    datanames,
    function(x) eventReactive(trigger_data(), datasets$get_data(x, filtered = TRUE)),
    simplify = FALSE
  )

  hashes <- calculate_hashes(datanames, datasets)

  teal.data::new_teal_data(
    data,
    ...
  )


```

Reply:
So what you're saying is `trigger_data` **shouldn't** be used in `.datasets_to_data`? And `.datasets_to_data` should be called in response to `trigger_data`?

Reply:
Yes, I think [this comment](https://github.com/insightsengineering/teal/pull/924#discussion_r1342628538) is exactly what we need. `.datasets_to_data` shouldn't use `trigger_data`

Reply:
Got it :+1: